core/deployment/src/main/java/io/quarkus/deployment/steps/LocaleProcessor.java

```diff
@@ -0,0 +1,151 @@
+        resources.produce(new NativeImageResourceBundleBuildItem("sun.util.resources.LocaleNames"));
+        resources.produce(new NativeImageResourceBundleBuildItem("sun.util.resources.CurrencyNames"));
+        //Adding sun.util.resources.TimeZoneNames is not necessary.
+    }
+
+    @BuildStep(onlyIf = { NativeBuild.class, NonEnglishLocale.class })
+    ReflectiveClassBuildItem setupReflectionClasses() {
+        return new ReflectiveClassBuildItem(false, false,
+                "sun.util.resources.provider.SupplementaryLocaleDataProvider",
+                "sun.util.resources.provider.LocaleDataProvider");
+    }
+
+    @BuildStep(onlyIf = { NativeBuild.class, NonEnglishLocale.class })
+    void servicesResource(BuildProducer<NativeImageResourceBuildItem> nativeImageResources,
+            BuildProducer<GeneratedResourceBuildItem> generatedResources) {
+        final String r1 = "META-INF/services/sun.util.resources.LocaleData$SupplementaryResourceBundleProvider";
+        final String r2 = "META-INF/services/sun.util.resources.LocaleData$CommonResourceBundleProvider";
+        nativeImageResources.produce(new NativeImageResourceBuildItem(r1, r2));
+        generatedResources.produce(new GeneratedResourceBuildItem(r1,
+                "sun.util.resources.provider.SupplementaryLocaleDataProvider".getBytes(StandardCharsets.UTF_8)));
+        generatedResources.produce(new GeneratedResourceBuildItem(r2,
+                "sun.util.resources.provider.LocaleDataProvider".getBytes(StandardCharsets.UTF_8)));
+    }
+
+    /**
+     * We activate additional resources in native-image executable only if user opts
+     * for something else than US English.
+     */
+    static final class NonEnglishLocale implements BooleanSupplier {
+        private final NativeConfig nativeConfig;
+        private final LocalesBuildTimeConfig localesBuildTimeConfig;
+
+        public NonEnglishLocale(NativeConfig nativeConfig, LocalesBuildTimeConfig localesBuildTimeConfig) {
+            this.nativeConfig = nativeConfig;
+            this.localesBuildTimeConfig = localesBuildTimeConfig;
+        }
+
+        @Override
+        public boolean getAsBoolean() {
+            return (nativeConfig.userLanguage.isPresent() && nonEnglishLocale(nativeConfig.userLanguage.get()))
+                    ||
+                    (nativeConfig.userCountry.isPresent() && nonEnglishLocale(nativeConfig.userCountry.get()))
+                    ||
+                    nonEnglishLocale(localesBuildTimeConfig.defaultLocale)
+                    ||
+                    localesBuildTimeConfig.locales.stream().anyMatch(LocaleProcessor::nonEnglishLocale);
+        }
+    }
+
+    public static boolean nonEnglishLocale(Locale locale) {
+        return (!locale.getLanguage().isEmpty() && !"en".equals(locale.getLanguage()))
+                || (!locale.getCountry().isEmpty() && !"US".equals(locale.getCountry()));
```

Comment:
`en_US` is always the default locale included in the native executable? My guess would have been that it's the default locale from your system instead (which happens to be `en_US` in your case).
So that this should be `nonDefaultLocale`. But I haven't looked at the GraalVM code so I might be incorrect.